pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -108,4 +112,154 @@ public void testMultipleGauges() {
+
+    // Test timed table APIs
+    testMetrics.addTimedTableValue(tableName, timer, 6, TimeUnit.SECONDS);
+    final MetricName t1Metric = inspector.lastMetric();
+    Assert.assertEquals(inspector.getTimer(t1Metric).sum(), 6000);
+    testMetrics.addTimedTableValue(tableName, keyName, timer, 500, TimeUnit.MILLISECONDS);
+    final MetricName t2Metric = inspector.lastMetric();
+    Assert.assertEquals(inspector.getTimer(t2Metric).sum(), 500);
+
+    // Test timed value APIs
+    testMetrics.addTimedValue(timer, 40, TimeUnit.MILLISECONDS);
+    final MetricName t3Metric = inspector.lastMetric();
+    Assert.assertEquals(inspector.getTimer(t3Metric).sum(), 40);
+    testMetrics.addTimedValue(keyName, timer, 3, TimeUnit.MILLISECONDS);
+    final MetricName t4Metric = inspector.lastMetric();
+    Assert.assertEquals(inspector.getTimer(t4Metric).sum(), 3);
+
+    // Remove added timers and verify the metrics registry is now empty
+    Assert.assertEquals(testMetrics.getMetricsRegistry().allMetrics().size(), 4);
+    testMetrics.removeTableTimer(tableName, timer);
+    testMetrics.removeTimer(t2Metric.getName());
+    testMetrics.removeTimer(t3Metric.getName());
+    testMetrics.removeTimer(t4Metric.getName());
+    Assert.assertTrue(testMetrics.getMetricsRegistry().allMetrics().isEmpty());
+  }
+
+  /**
+   * Tests the {@link AbstractMetrics} APIs relating to metered metrics
+   */
+  @Test
+  public void testMeteredMetrics() {
+    final ControllerMetrics testMetrics = buildTestMetrics();
+    final MetricsInspector inspector = new MetricsInspector(testMetrics.getMetricsRegistry());
+    final String tableName = "tbl_testMeteredMetrics";
+    final String keyName = "keyName";
+    final ControllerMeter meter = ControllerMeter.CONTROLLER_INSTANCE_POST_ERROR;
+    final ControllerMeter meter2 = ControllerMeter.CONTROLLER_PERIODIC_TASK_ERROR;
+
+    // Holder for the most recently seen Metric
+    final MetricName[] currentMetric = new MetricName[1];
+
+    // When a new metric is expected we'll call this lambda to assert the metric was created, and update currentMetric
+    final Runnable expectNewMetric = () -> {
+      Assert.assertNotEquals(inspector.lastMetric(), currentMetric[0]);
+      currentMetric[0] = inspector.lastMetric();
+    };
+
+    // Lambda to verify that the latest metric has the expected value, and its creation was expected
+    final IntConsumer expectMeteredCount = expected -> {
+      Assert.assertEquals(inspector.getMetered(currentMetric[0]).count(), expected);
+      Assert.assertEquals(currentMetric[0], inspector.lastMetric());
+    };
+
+    // Test global meter APIs
+    testMetrics.addMeteredGlobalValue(meter, 5);
+    expectNewMetric.run();
+    expectMeteredCount.accept(5);
+    testMetrics.addMeteredGlobalValue(meter, 4, testMetrics.getMeteredValue(meter));
+    expectMeteredCount.accept(9);
+
+    // Test meter with key APIs
+    testMetrics.addMeteredValue(keyName, meter, 9);
+    expectNewMetric.run();
```

Comment:
let's run `expectMeteredCount` for `addMeteredValue(keyName, meter, 9)` as well

Reply:
Done.